tests/test_api_v1.py

```diff
@@ -189,6 +189,40 @@ def test_fetch_paginated_requests(
     assert type(fetched_requests[0]['dependencies']) == list
 
 
+def test_create_request_filter_state(app, auth_env, client, db):
+    """Test that requests can be filtered by state."""
+    repo_template = 'https://github.com/release-engineering/retrodep{}.git'
+    # flask_login.current_user is used in Request.from_json, which requires a request context
+    with app.test_request_context(environ_base=auth_env):
+        for i in range(10):
+            # Make a request in 'in_progress' state
+            data = {
+                'repo': repo_template.format(i),
+                'ref': 'c50b93a32df1c9d700e3e80996845bc2e13be848',
+                'pkg_managers': ['gomod'],
+            }
+            request = Request.from_json(data)
+            db.session.add(request)
+            # Make a request in 'complete' state
+            data_complete = {
+                'repo': repo_template.format(i),
+                'ref': 'e1be527f39ec31323f0454f7d1422c6260b00580',
+                'pkg_managers': ['gomod'],
+            }
+            request_complete = Request.from_json(data_complete)
+            request_complete.add_state('complete', 'Completed successfully')
+            db.session.add(request_complete)
+    db.session.commit()
+
+    rv = client.get('/api/v1/requests?state=complete')
```

Comment:
All your requests are in the `complete` state, so this test doesn't actually verify if the filter was applied. Could you please change your test environment to have at least one request in a different state?

Reply:
The requests I tested are half in progress and half complete. They start out as in progress and then for the ones using the `request_complete` variable, they get a `complete` state added to them on line 213. (Unless I am using them wrong, which may be so!) What I intended to do was make 2 requests for every loop, for a total of 20, and then half of them would be those 10 fetched requests I assert later on.

Reply:
I'm sorry. I misread your code. This test is accurate, but I'd still like to see the other comment about making it more efficient addressed.